### PR TITLE
Lps 159384 Reuse logic from PortalUtil in BasePortletLayoutFinder

### PR DESCRIPTION
**Pull request by JavierMoral**
Hi,
this pull fixes https://issues.liferay.com/browse/LPS-159384

**Reproduction steps:**

- Be sure that your bundle is connected to a fake SMTP server like [fakeSMTP](http://nilhcem.com/FakeSMTP/) or [MockMock](https://github.com/tweakers/MockMock) in order to view the email notifications
- Login with Test user
- Create new folder in Documents and Media
- Create a user 'usr1' and add it as a member of the site 'Guest'
- Create a content page in 'Guest' site
- Add the Documents and media widget to the page created in step 5
- Login or impersonate 'usr1' user
- Subscribe to the folder created in step 3 in order to receive notifications when a new upload is done inside this folder
- Login with Test user
- Add new file to this folder to send new notification to subscribed user

 

 **Expected behavior:**

Customer expect to receive the link using friendlyURL of the page
 

 **Actual behavior:**

The link sent in the email notification is generated for control panel (http://localhost:8080/group/guest/~/control_panel/)
 

**Additional information:**
Class com.liferay.portal.kernel.portlet.BasePortletLayoutFinder implemented its own logic to find portlets added to layouts and it was only compatible with widget pages. My idea was to use already existing logic in PortalUtil.getPlidFromPortletId to achieve this as it is also compatible with content pages.

I also had to move method getAllNonembeddedPortlets from PortlImpl to LayoutTypePortletImpl because all existing methods in this class are only compatible with widget pages so I needed a method compatible also with content pages. It also made sense looking at its logic.

### CODE REVIEW

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
ci:test:sf

File: portal-kernel/src/com/liferay/portal/kernel/portlet/BasePortletLayoutFinder.java [hunk 1]
```diff
@@ -120,12 +116,14 @@ public Result find(ThemeDisplay themeDisplay, long groupId)
 	protected String getPortletId(
 		LayoutTypePortlet layoutTypePortlet, String portletId) {
 
-		for (String curPortletId : layoutTypePortlet.getPortletIds()) {
+		for (Portlet curPortletId :
```
Comment (adolfopa) [inline portal-kernel/src/com/liferay/portal/kernel/portlet/BasePortletLayoutFinder.java:119]: `curPortletId` is no longer an ID, it is a `Portlet`. Can you rename the variable to match the type?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: portal-kernel/src/com/liferay/portal/kernel/portlet/BasePortletLayoutFinder.java (hunk 1 above)
Comment (JavierMoral) [reply portal-kernel/src/com/liferay/portal/kernel/portlet/BasePortletLayoutFinder.java:119]: Sure, I fixed the commit and resent in #4341

**Comment by JavierMoral** [comment]
ci:close

**Pull request closed without merging**